packages/payment-processor/src/payment/utils.ts

```diff
@@ -306,6 +313,65 @@ export function validateConversionFeeProxyRequest(
   }
 }
 
+/**
+ * Validates the parameters for an ERC20 Fee Proxy payment before payment
+ * @param request to validate
+ * @param amount optionally, the custom amount to pay
+ * @param feeAmountOverride optionally, the custom fee amount
+ * @param signerOrProvider
+ */
+export async function validatePayERC20TransferableReceivable(
+  request: ClientTypes.IRequestData,
+  signerOrProvider: providers.Provider | Signer,
+  amount?: BigNumberish,
+  feeAmountOverride?: BigNumberish,
+): Promise<void> {
+  const receivableTokenId = await getReceivableTokenIdForRequest(request, signerOrProvider);
+
+  if (receivableTokenId.isZero()) {
+    throw new Error(
+      'The receivable for this request has not been minted yet. Please check with the payee.',
+    );
+  }
+
+  validateERC20TransferableReceivable(request, amount, feeAmountOverride);
+}
+
+/**
+ * Validates the parameters for an ERC20 Fee Proxy payment.
```

Comment:
Incorrect header comment.

```suggestion
 * Validates the parameters for an ERC20 Transferable Receivable Payment or Mint.
```